frigate/config.py

```diff
@@ -890,7 +834,7 @@ class FrigateConfig(FrigateBaseModel):
         default_factory=ModelConfig, title="Detection model configuration."
     )
     detectors: Dict[str, DetectorConfig] = Field(
-        default={name: DetectorConfig(**d) for name, d in DEFAULT_DETECTORS.items()},
+        default=parse_obj_as(Dict[str, DetectorConfig], DEFAULT_DETECTORS),
```

Comment:
I think this can just be DEFAULT_DETECTORS, no need for parse_obj_as in the default

Reply:
all the above suggestions are implemented. I also added some docs on how to provide a custom implementation that I'm pretty sure will need some word-smithing.